prow/config/jobs/envoy.yaml

```diff
@@ -0,0 +1,21 @@
+org: istio
+repo: envoy
+branches: [master, release-1.4, release-1.5]
+image: piotrsikora/envoy-build-ubuntu@sha256:e2775df4cf57e86920ca39886a877b4ccc12dbcfaa2a7c2a804f4cb83c797952
+
+jobs:
+- name: test-asan
```

Comment:
Is there any reason why do we test only ASan?

Reply:
There is no reason other than ambiguity. This PR also includes setting up the minimum files and Prow config. We can trivially and incrementally add more tests. It is not clear to me what tests we want: https://github.com/istio/test-infra/issues/2385.

Reply:
`bazel.release` and `bazel.tsan`. It's fine if you want to add them in a follow-up, but it seems like it would be easy to add (and verify they work) in this PR as well?

Reply:
It looks good otherwise, so let me know if you prefer to merge it as-is now. I didn't approve it yet to prevent bot from auto-merging it.

Reply:
I will add them.

Reply:
@PiotrSikora 

Are these flakes or legitimate errors?

https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_envoy/152/test-asan_envoy/1
https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_envoy/152/test-tsan_envoy/1
https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_envoy/152/test-release_envoy/1